File: api-cvc/src/tr/gov/tubitak/uekae/esya/api/cvc/CVCChainInfoProvider.java
package tr.gov.tubitak.uekae.esya.api.cvc;

import tr.gov.tubitak.uekae.esya.api.asn.cvc.ENonSelfDescCVCwithHeader;

import java.security.PublicKey;
import java.util.ArrayList;
import java.util.List;

/**
 <b>Author</b>    : zeldal.ozdemir <br>
 <b>Project</b>   : MA3   <br>
 <b>Copyright</b> : TUBITAK Copyright (c) 2006-2011 <br>
 <b>Date</b>: 11/17/11 - 2:25 PM <p>
 <b>Description</b>: <br>
 */
public class CVCChainInfoProvider implements ICVCChainInfoProvider {
    private List<ENonSelfDescCVCwithHeader> subCvcAuthorities = new ArrayList<ENonSelfDescCVCwithHeader>();
    private PublicKey cvcRootPublicKey;
    private ENonSelfDescCVCwithHeader rootCVCAuthority;

    public CVCChainInfoProvider() {
    }

    public List<ENonSelfDescCVCwithHeader> getSubCvcAuthorities() {
        return subCvcAuthorities;
    }

    public PublicKey getRootCVCAuthPublic() {
        return cvcRootPublicKey;
    }

    public void setRootCVCAuthority(ENonSelfDescCVCwithHeader rootCVCAuthority){
        this.rootCVCAuthority = rootCVCAuthority;
    }

    public void addSubCVCAuthority(ENonSelfDescCVCwithHeader subCvcAuthority){
        subCvcAuthorities.add(subCvcAuthority);
    }

    public void setRootCVCPublicKey(PublicKey cvcRootPublicKey) {
        this.cvcRootPublicKey = cvcRootPublicKey;
    }

    public ENonSelfDescCVCwithHeader getRootCVCAuthority() {
        return rootCVCAuthority;
    }
}

File: api-cvc/csharp/src/tr/gov/tubitak/uekae/esya/api/cvc/CVCChainInfoProvider.cs
using System.Collections.Generic;
using tr.gov.tubitak.uekae.esya.api.asn.cvc;
using tr.gov.tubitak.uekae.esya.api.crypto.provider.core.baseTypes;

namespace tr.gov.tubitak.uekae.esya.api.cvc
{
    public class CVCChainInfoProvider : ICVCChainInfoProvider
    {
        private List<ENonSelfDescCVCwithHeader> subCvcAuthorities = new List<ENonSelfDescCVCwithHeader>();
        private IPublicKey cvcRootPublicKey;
        private ENonSelfDescCVCwithHeader rootCVCAuthority;

        public CVCChainInfoProvider()
        {
        }

        public List<ENonSelfDescCVCwithHeader> getSubCvcAuthorities()
        {
            return subCvcAuthorities;
        }

        public IPublicKey getRootCVCAuthPublic()
        {
            return cvcRootPublicKey;
        }

        public void setRootCVCAuthority(ENonSelfDescCVCwithHeader rootCVCAuthority)
        {
            this.rootCVCAuthority = rootCVCAuthority;
        }

        public void addSubCVCAuthority(ENonSelfDescCVCwithHeader subCvcAuthority)
        {
            subCvcAuthorities.Add(subCvcAuthority);
        }

        public void setRootCVCPublicKey(IPublicKey cvcRootPublicKey)
        {
            this.cvcRootPublicKey = cvcRootPublicKey;
        }

        public ENonSelfDescCVCwithHeader getRootCVCAuthority()
        {
            return rootCVCAuthority;
        }
    }
}
